File: common/src/main/java/org/sandboxpowered/silica/server/SilicaServer.java
package org.sandboxpowered.silica.server;

import akka.actor.typed.ActorRef;
import org.sandboxpowered.api.server.Server;
import org.sandboxpowered.silica.StateManager;
import org.sandboxpowered.silica.command.Commands;
import org.sandboxpowered.silica.resources.ClasspathResourceLoader;
import org.sandboxpowered.silica.resources.ResourceManager;
import org.sandboxpowered.silica.resources.ResourceType;
import org.sandboxpowered.silica.world.SilicaWorld;

import java.security.KeyPair;
import java.security.KeyPairGenerator;
import java.security.NoSuchAlgorithmException;
import java.util.Random;

public abstract class SilicaServer implements Server {
    private final KeyPair encryptionKeyPair;
    private final byte[] verificationArray = new byte[4];
    private final Random serverRandom = new Random();
    private final Commands commands;
    public ServerProperties properties;
    public ResourceManager dataManager;

    public SilicaServer() {
        try {
            KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
            keyPairGenerator.initialize(1024);
            encryptionKeyPair = keyPairGenerator.generateKeyPair();
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
        serverRandom.nextBytes(verificationArray);
        commands = new Commands();
        dataManager = new ResourceManager(ResourceType.DATA);
        dataManager.add(new ClasspathResourceLoader());
    }

    public Commands getCommands() {
        return commands;
    }

    public byte[] getVerificationArray() {
        return verificationArray;
    }

    public KeyPair getKeyPair() {
        return encryptionKeyPair;
    }

    public abstract StateManager getStateManager();

    public abstract ActorRef<SilicaWorld.Command> getWorld();
}

File: common/src/main/kotlin/org/sandboxpowered/silica/server/SilicaServer.kt
package org.sandboxpowered.silica.server

import com.mojang.authlib.minecraft.MinecraftSessionService
import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService
import org.sandboxpowered.silica.api.server.Server
import org.sandboxpowered.silica.command.Commands
import org.sandboxpowered.silica.resources.ClasspathResourceLoader
import org.sandboxpowered.silica.resources.ResourceManager
import org.sandboxpowered.silica.resources.ResourceType
import java.net.Proxy
import java.util.*

abstract class SilicaServer : Server {
    private val serverRandom = Random()
    val commands: Commands = Commands()
    var dataManager: ResourceManager = ResourceManager(ResourceType.DATA)
    override val sessionService: MinecraftSessionService =
        YggdrasilAuthenticationService(Proxy.NO_PROXY).createMinecraftSessionService()

    abstract fun shutdown()

    init {
        dataManager.add(ClasspathResourceLoader("Silica", arrayOf("silica", "minecraft")))
    }
}